Flesh out the Responsive Web Design Stub (#39663)
freeCodeCamp/freeCodeCamp: client/src/pages/learn/responsive-web-design/index.md

---
## Introduction to Responsive Web Design

Welcome to Responsive Web Design!

Let's first take a look at what is in the scope of Responsive Web Design (and what is not).

> The Web part of Responsive Web Design means that the content you are creating is designed to be delivered over the web, to people's browsers. To do this, you will need to learn the language that browsers use to describe web pages: HTML (Hypertext Markup Language) for content, and CSS (Cascading Style Sheets) for Designing.

Reviewer: ```suggestion
The Web part of Responsive Web Design means that the content you are creating is designed to be delivered over the web to users' browsers. To do this, you will need to learn the language that browsers use to describe web pages: HTML (Hypertext Markup Language) for content, and CSS (Cascading Style Sheets) for design.
```

> That brings us to design. Web Design focuses on how websites should look and behave to the end user. There are plenty of additional aspects to websites, including how to host a website so people can access it; how to protect your websites and store your users' data, and the ethos behind what should go on a website in the first place. These are somewhat harder to teach, but do require a good understanding of the capabilities of the web in modern contexts. 

Reviewer: ```suggestion
That brings us to design. Web Design focuses on how websites should look and behave to the end user. There are plenty of additional aspects to websites, including how to host a website so people can access it, how to protect your websites and store your users' data, and the ethos behind what should go on a website in the first place. These are somewhat harder to teach, but do require a good understanding of the capabilities of the web in modern contexts. 
```

> And what of the last word - Responsive? Well, that brings us to the idea of modern contexts. In today's world, websites are expected to be just as accessible on laptops, tablets and phones. If you compare your user experience on a document-editing program on a computer versus a document-editing program on a mobile, the features on mobile are likely to be far more constrained. This applies in some way to websites as well, but developers have found interesting ways to adapt to the mobile experience, not in the least by harnessing powerful features built for flexible (a.k.a responsive) web displays such as `css-flexbox` and `css-grid`.

Reviewer 2: ```suggestion
And what of the last word - Responsive? Well, that brings us to the idea of modern contexts. In today's world, websites are expected to be just as accessible on laptops, tablets and phones. If you compare your user experience on a document-editing program on a computer versus a document-editing program on a mobile, the features on mobile are likely to be far more constrained. This applies in some way to websites as well, but developers have found interesting ways to adapt to the mobile experience, not in the least by harnessing powerful features built for flexible (a.k.a responsive) web displays such as CSS `flexbox` and CSS `grid`.
```

@Manish-Giri 

Expanding on what your review comment means in terms of changes? If this is OK should I commit this?

Reviewer 3: @raisedadead Yes, looks good! Please commit.

> And what of the last word - Responsive? Well, that brings us to the idea of modern contexts. In today's world, websites are expected to be just as accessible on laptops, tablets and phones. If you compare your user experience on a document-editing program on a computer versus a document-editing program on a mobile, the features on mobile are likely to be far more constrained. This applies in some way to websites as well, but developers have found interesting ways to adapt to the mobile experience, not in the least by harnessing powerful features built for flexible (a.k.a responsive) web displays such as CSS `flexbox` and CSS `grid`.

Reviewer 4: > And what of the last word - Responsive? 

Not sure what is meant here by "last word".  That would actually be the first word in the phrase "Responsive Web Design".  Also, let's avoid using hyphen between "word" and "Responsive"?  I think this sentence would read better as:

> And what about the word Responsive?

Reviewer 4: @acenturyandabit  Can you respond to my request?

Author: Last as in "the one we haven't looked at yet in the preceding paragraphs". But it's a minor change, I'm sure our readers won't be thrown too much by it. Happy to go with your suggestion

Reviewer 4: I would feel better about approving the PR with the verbiage I suggested above.

Author: Alrighty, done!

> And what of the last word - Responsive? Well, that brings us to the idea of modern contexts. In today's world, websites are expected to be just as accessible on laptops, tablets and phones. If you compare your user experience on a document-editing program on a computer versus a document-editing program on a mobile, the features on mobile are likely to be far more constrained. This applies in some way to websites as well, but developers have found interesting ways to adapt to the mobile experience, not in the least by harnessing powerful features built for flexible (a.k.a responsive) web displays such as `css-flexbox` and `css-grid`.

Reviewer 3: Do we really need to use the `css-` prefix with Flexbox and grid - 

> web displays such as `css-flexbox` and `css-grid`

It appears like they are literally called `css-flexbox` and `css-grid`. Maybe removing the `-` should fix it.
